internal/internal_task_pollers.go

```diff
@@ -1245,6 +1246,21 @@ func convertActivityResultToRespondRequest(
 			}
 		}
 		if errors.Is(err, context.Canceled) {
+			// Cancelation due to worker shutdown should be wrapped in an
+			// Application Error, so we don't waste time waiting for the
+			// activity to timeout from the server side.
+			if canceledFromWorkerShutdown {
```

Comment:
There is already code to handle this in the SDK, look at `cancelAllowed` we shouldn't add another flag, just fix the broken flag